src/include/rejson_api.h

```diff
@@ -65,6 +71,54 @@ typedef struct RedisJSONAPI_V1 {
 
 } RedisJSONAPI_V1;
 
+typedef struct RedisJSONAPI_V2 {
```

Comment:
Put `RedisJSONAPI_V1` as the first element here instead of copy it

Reply:
But it will not be flat, and I wanted to be able to call v1 APIs indifferently whether v1 or v2 API was actually loaded.

Reply:
When we go C++ we can inherit, in the meantime we can play around with macros.

Reply:
You can return v2 api even if v2 is requested, v2 is just an extention on v1..